### ISSUE TITLE
MVP2 reasoning seems off

### ISSUE THREAD

**Issue by TranslatorIssueCreator** [NCATSTranslator/Feedback#755]
## Steps to reproduce:
what gene may be upregulated by acetylcholine?
search for LAMA3

**Comment by sandrine-muller** [comment]
from improving agent:
![image](https://github.com/NCATSTranslator/Feedback/assets/153299800/19fd0e4f-18fc-465e-a015-9161770f8a83)

the predicate chain is the following:
LAMA3 `Regulates Increased Activity Or Abundance Of` CDK6 `Has Increased Expression Affected By` Acetylcholine

I do not understand why LAMA3 is the answer to the question while CDK6 is not? 

there are in the top ranked a bunch of genes as valid answers where CDK6 in an intermediate node

**Comment by sierra-moxon** [comment]
reviewing the ticket in triage:
@sandrine-muller - is LAMA3 a valid answer to this query?

@suihuang-ISB @brettasmi - is this something you could look into?  The question asks for a chemical and the interface seems to be returning a gene?

**Comment by Genomewide** [comment]
I don't think it is a valid answer to the query (IMO).  I went to ARAX UI to make sure we were not messing up the predicates some how, and this is just bad logic. It is from improving agent and I don't think it should be an answer that is coming back. I don't know who to tag on that team? @sierra-moxon do you know who to ask about this?

![image](https://github.com/NCATSTranslator/Feedback/assets/55069176/2efb7a3c-eb7d-4d65-a1bb-d6553e7e5fcd)

**Comment by mbrush** [comment]
Note as an aside that the composed predicates in the support path Sandrine pasted above are not reading correctly - likely because of the issue I documented [here](https://github.com/NCATSTranslator/Feedback/issues/744) concerning KPs not including all the qualifiers necessary for the UI to compose the right edge label. 

**LAMA3** `regulates increased activity or abundance Of` **CDK6** `has increased expression affected by` **Acetylcholine**

should be:

**LAMA3** `CAUSES increased activity or abundance of` **CDK6** `has increased expression CAUSED by` **Acetylcholine**

**Comment by sandrine-muller** [comment]
I agree with Andy, I think CDK6 should be an answer but not the other genes like LAMA3.

**Comment by brettasmi** [comment]
I'll take a look at this; thanks for reporting. It looks like the directionality of the relationships in our query logic might not be coming through correctly.

**Comment by suihuang-ISB** [comment]
Hi @brettasmi - yep, I think so too. Part of the problem is from enforcing 2 hops (gene1/protein)--[Xregulates]-->(gene2)--Xregulates]-->(gene3)  and then getting something wrong with the directionality of the edge.

But independently, I was not able to find in SPOKE the edges CDK6 - Lama3, and that between Acetylcholine and CDK6. (Obviously everything that stimulates proliferation will upregulate CDK6 anyway - so I do not see the point of the query here).

@mbrush :  Not sure if t is is a biolink/predicate qualifier issue. At least it should not be. I hope not.  The direction of edges is so profound and flows into reasoning, even into the actual grpah query, so early in the answering process that it should not be something that the UI "composes" based on qualifiers.

**Comment by sandrine-muller-research** [comment]
Retested [today](https://ui.test.transltr.io/main/results?l=Acetylcholine&i=CHEBI:15355&t=3&r=0&q=29528aec-3052-4b95-b9e9-33e1037962de)

The correct qualified predicates appear here properly:
<img width="467" alt="image" src="https://github.com/user-attachments/assets/aa4457ca-2640-413a-8c19-b325bfdf8bae">

CDK6 is still not returned by any ARA (while the edge is present) but the LAMA3 2 hops shows up.
